### PR TITLE
Explore: Remove legend formatting when switching from panel to Explore

### LINKED ISSUE #14393: Explore: Jumping to Explore keeps target options like labelFormat
1. Prometheus datasource, Explore enabled
1. Create dashboard panel with prometheus query `promhttp_metric_handler_requests_total`
1. Set a label format: `{{code}}`
1. Notice the nicely formatted legend labels
1. Jump to Explore via the panel menu
1. Notice how the formatted labels are preserved
1. Change query to `up`

Result: legend label is now `code`, which makes no sense.

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
In Explore, we don't have legend formatter and we don't want to keep legend formatting as we can't later change it. 

**Which issue(s) this PR fixes**:
Fixes https://github.com/grafana/grafana/issues/14393

### CODE REVIEW

**Review by besartberisha — APPROVED** [review]
Looks good to me, I noticed that CI failed so I'd check that again but the fix seems clear also appreciated the comment there to explain why it was `_omit`ed.

**Comment by ivanahuckova** [comment]
@besartberisha yeah, it was failing at **build-enterprise-plugins** which this PR has nothing to do with. So I merged the current master, which will probably fix the issue. 🙂

File: public/app/core/utils/explore.ts [hunk 1]
```diff
@@ -69,7 +69,11 @@ export interface GetExploreUrlArguments {
 export async function getExploreUrl(args: GetExploreUrlArguments): Promise<string | undefined> {
   const { panel, panelTargets, panelDatasource, datasourceSrv, timeSrv } = args;
   let exploreDatasource = panelDatasource;
-  let exploreTargets: DataQuery[] = panelTargets;
+
+  /** In Explore, we don't have legend formatter and we don't want to keep
+   * legend formatting as we can't change it
+   */
+  let exploreTargets: DataQuery[] = panelTargets.map(t => _.omit(t, 'legendFormat'));
```
Comment (aocenas) [inline public/app/core/utils/explore.ts:76]: Can we add a test for this change?

Comment (ivanahuckova) [reply public/app/core/utils/explore.ts:76]: Added 🙂

**Pull request merged**